cpp/src/arrow/testing/gtest_util.cc

```diff
@@ -769,10 +770,8 @@ class GatingTask::Impl {
   }
 
   Status Unlock() {
-    {
-      std::lock_guard<std::mutex> lk(mx_);
-      unlocked_ = true;
-    }
+    std::lock_guard<std::mutex> lk(mx_);
```

Comment:
Looks it's better to release the lock before calling cv.notify_all()?
https://en.cppreference.com/w/cpp/thread/condition_variable/notify_all
```
The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s); in fact doing so is a pessimization, since the notified thread would immediately block again, waiting for the notifying thread to release the lock.
```

Reply:
Indeed, but this is a test util so I'd favor simplicity.